Kudu.Core/Deployment/DeploymentHelper.cs

```diff
@@ -75,5 +78,45 @@ public static void PurgeZipsIfNecessary(string sitePackagesPath, ITracer tracer,
                 }
             }
         }
+
+        public static async Task<HttpContent> PerformGetAsync(HttpClient client, Uri uri, ITracer tracer)
+        {
+            int retries = 3;
+            int delay = 200;
+            HttpResponseMessage response = null;
+            while (true)
+            {
+                using (tracer.Step("Trying to make a GET request to {0}", uri.AbsoluteUri))
+                {
+                    try
+                    {
+                        response = await client.GetAsync(uri);
+                        response.EnsureSuccessStatusCode();
+                        return response.Content;
+                    }
+                    catch (Exception ex)
+                    {
+                        if (--retries <= 0)
+                        {
+                            tracer.TraceError(ex, "Could not make a successful GET request to {0}", uri.AbsoluteUri);
+                            throw ex;
+                        }
+                        Thread.Sleep(delay);
```

Comment:
use `await Task.Delay(delay);` instead of `Thread.Sleep` since the latter will block a thread for the sleep.